Core/include/Acts/EventData/SeedContainer2.hpp

```diff
@@ -156,7 +156,7 @@ class SeedContainer2 {
     IndexType m_index{};
 
     friend bool operator==(const SeedIterator &a, const SeedIterator &b) {
-      return a.m_index == b.m_index && a.m_container == b.m_container;
+      return a.m_index == b.m_index;
```

Comment:
We're assuming comparing iterators from different containers won't be an issue?

Reply:
I would argue that this is the callers responsibility. I believe STL does the same - even though that should not mean we have to do it. But I think it would be more reasonable to throw or assert in this case because someone is doing something terribly wrong...